src/rest/rest_devices.c

```diff
@@ -306,18 +401,22 @@ int rest_devices_put_cb(const ulfius_req_t *req, ulfius_resp_t *resp, void *cont
         ulfius_set_empty_body_response(resp, 400);
         goto exit;
     }
-    if ((json_object_get(jdevice, "psk") == NULL) || (json_object_get(jdevice, "psk_id") == NULL))
+
+    j_string = json_object_get(jdevice, "name");
+    if (j_string == NULL)
     {
         ulfius_set_empty_body_response(resp, 400);
         goto exit;
     }
 
-    json_t *jstring = json_string(id);
-    json_object_set_new(jdevice, "uuid", jstring);
+    name = json_string_value(j_string);
+    if (name == NULL)
+    {
+        ulfius_set_empty_body_response(resp, 400);
```

Comment:
To increase code readability and implementation clarity, use status codes defined in `src/http_codes.h`.